server/src/test/java/org/elasticsearch/threadpool/ThreadPoolTests.java

```diff
@@ -238,4 +239,41 @@ public void testInheritContextOnSchedule() throws InterruptedException {
             terminate(threadPool);
         }
     }
+
+    public void testSchedulerWarnLogging() throws Exception {
+        final ThreadPool threadPool = new TestThreadPool("test",
+                Settings.builder().put(ThreadPool.SLOW_SCHEDULER_TASK_WARN_THRESHOLD_SETTING.getKey(), "10ms").build());
+        final Logger logger = LogManager.getLogger(ThreadPool.class);
+        final MockLogAppender appender = new MockLogAppender();
+        appender.start();
+        try {
+            Loggers.addAppender(logger, appender);
+            appender.addExpectation(new MockLogAppender.SeenEventExpectation(
+                    "expected warning for slow task",
+                    ThreadPool.class.getName(),
+                    Level.WARN,
+                    "execution of [slow-test-task] took [*ms] which is above the warn threshold of [10ms]"));
+            final Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        TimeUnit.MILLISECONDS.sleep(200L);
```

Comment:
We use the cached time thread to measure the elapsed time, which defaults to a resolution of 200ms, so even a little bit of jitter in the timings of things could cause this test to fail.

I would prefer to have a short sleep in a loop that runs until `threadPool.relativeTimeInNanos()` has moved at least 10ms, because that's what we're actually trying to await.

Reply:
Right ... was using `System.nanoTime()` when I wrote this but then moved to using cached time and forgot all about it :)